### PR TITLE
Add GitHub OIDC CloudFormation and update READMEs

### PR DESCRIPTION
**Pull request by ben-harvey**
This change adds a CloudFormation template to create GitHub OIDC resources, to be used by teams who prefer it for IaC.  Since the template will live alongside the Terraform module in the `github-oidc` folder, I restructured the READMEs to have all general GitHub OIDC information in the top-level README, and CloudFormation/Terraform-specific info in the READMEs for the respective sub-folders. 

Note that this introduces a breaking change to the path for the Terraform module, so any consumers need to update the module source reference. Currently, there are only internal consumers:
 - the `mac-fc-infra` repo
   - I'll create a PR there for the update once this is merged 
 -  a feature branch in the `macbis-api-gateway` repo
    - I'll let Henry know about this

Testing:
I applied the template to our test environment and ran a test workflow to confirm access to allowed AWS actions and lack of access to denied AWS actions

### CODE REVIEW

File: github-oidc/cloudformation/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+The permissions policy is defined by two optional parameters:
+
+- `GitHubActionsAllowedAWSActions` is a comma-separated list of strings that are comprised of an AWS service and an allowed action. See the [IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_action.html) for examples. The service/action pairs listed will be granted access to all resources (`"Resource": "*"`).
+- `ManagedPolicyARNs` is a comma-separated list of IAM policy ARNs to attach to the OIDC role. These can be AWS-managed or custom-managed policy ARNs.
+
+### Steps
+
+1. For each AWS environment, create a parameters JSON file with the values needed for that environment. An [example file](./parameters-example.json) is provided.
+
+2. Deploy the OIDC resources in each AWS environment using [the provided CloudFormation template](.github/oidc/github-actions-oidc-template.yml) and the AWS CLI:
```
Comment (yonassrobi) [inline github-oidc/cloudformation/README.md:20]: The template link seems to be broken here.

File: github-oidc/README.md [hunk 2]
```diff
@@ -1,63 +1,85 @@
...
 
-`github-oidc/main.tf`
+## IAM Role
 
-```hcl
-  module "github-actions-runner-aws" {
-    source = "github.com/CMSgov/github-actions-runner-aws//github-oidc" # double-slash denotes a sub-directory
+This is the role that the Actinos workflow assumes via the OIDC request. The permissions defined for this role dictate the scope of the short-term credentials that are generated and passed back to GitHub by AWS. The ARN of this role is output by each of the IaC methods listed, and is passed in to the [GitHub Action that configures AWS credentials for workflows](https://github.com/aws-actions/configure-aws-credentials#assuming-a-role). See the example below under "Using the OIDC Provider in a Workflow."
```
Comment (yonassrobi) [inline github-oidc/README.md:23]: Actions? (Typo)

File: github-oidc/README.md [hunk 3]
```diff
@@ -1,63 +1,85 @@
...
+  "repository_visibility": "public",
+  "repository_id": "376113068",
+  "actor_id": "25254258",
+  "actor": "{github username}",
+  "workflow": "Run security scans",
+  "head_ref": "",
+  "base_ref": "",
+  "event_name": "push",
+  "ref_type": "branch",
+  "job_workflow_ref": "{organization}/{repo}/.github/workflows/security.yml@refs/heads/bharvey-security-skip-build",
```
Comment (yonassrobi) [inline github-oidc/README.md:57]: I don't think it matters all that much but should this reference the main branch?

**Review by yonassrobi — COMMENTED** [review]
Looks great @ben-harvey. Awesome job in thoroughly documenting the OIDC implementation and providing step by step guide for the new cloud formation alternative. Added a couple of minor comments. At some point, if and when we move github-actions-runner-aws to Enterprise-CMCS org, do you think it would make sense to create a new repo for OIDC altogether?

File: github-oidc/README.md (hunk 3 above)
Comment (ben-harvey) [reply github-oidc/README.md:57]: Good call, I generalized a couple of other specific things as well.

**Comment by ben-harvey** [comment]
> At some point, if and when we move github-actions-runner-aws to Enterprise-CMCS org, do you think it would make sense to create a new repo for OIDC altogether?
Great idea!  It is a little awkward how this is shoehorned into the runner repo.

**Review by yonassrobi — APPROVED** [review]
LGTM!

**Pull request merged**